model/src/main/kotlin/OrtResult.kt

```diff
@@ -93,21 +97,28 @@ data class OrtResult(
             } ?: sortedSetOf<String>()
 
     /**
-     * Return all declared licenses associated to the projects / packages they occur in.
+     * Return all declared licenses associated to the projects / packages they occur in. If [omitExcluded] is set to
+     * true, excluded projects / packages are omitted from the result.
      */
     @Suppress("UNUSED") // This is intended to be mostly used via scripting.
-    fun collectAllDeclaredLicenses() =
+    fun collectAllDeclaredLicenses(omitExcluded: Boolean = false) =
             sortedMapOf<String, SortedSet<Identifier>>().also { licenses ->
+                val excludes = repository.config.excludes.takeIf { omitExcluded }
+
                 analyzer?.result?.let { result ->
-                    result.projects.forEach {
-                        it.declaredLicenses.forEach { license ->
-                            licenses.getOrPut(license) { sortedSetOf() } += it.id
+                    result.projects.forEach { project ->
+                        if (excludes?.isProjectExcluded(project) == false) {
+                            project.declaredLicenses.forEach { license ->
+                                licenses.getOrPut(license) { sortedSetOf() } += project.id
+                            }
                         }
                     }
 
-                    result.packages.forEach {
-                        it.pkg.declaredLicenses.forEach { license ->
-                            licenses.getOrPut(license) { sortedSetOf() } += it.pkg.id
+                    result.packages.forEach { (pkg, _) ->
+                        if (excludes?.isPackageExcluded(pkg.id, analyzer.result) == false) {
```

Comment:
You have to use `!= true` instead, because if `excludes` is `null` the above expression evaluates to `false`.